### PR TITLE
239 show vertical profile

### PR DESCRIPTION
**Pull request by vanmeeuwen**
The vertical profile works fine at least for CMEMS: http://localhost:8080/metoc/oceanographic/temperature/oceanographic_temperature_cmems/coordinates/59.19/17.25
D3D though doesn't show anything since the request id cannot be found in the seriesstore. I think this needs to be fixed in the backend and solved in another issue where the ids are retrieved from the backend differently than we do now with sequence numbers in subplots

### CODE REVIEW

File: src/components/TimeSeriesComponent/ElevationChart.vue [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+      if (!series.visibleInPlot) continue
+      if (!ids.includes(series.id)) {
+        this.addToChart(series)
+      }
+      const index = removeIds.findIndex((item) => {
+        return item === series.id
+      })
+      if (index >= 0) removeIds.splice(index, 1)
+    }
+    for (const id of removeIds) {
```
Comment (martinapippi) [inline src/components/TimeSeriesComponent/ElevationChart.vue:147]: use the more compact way: `removeIds.forEach(id => this.axis.removeChart(id))`

**Review by martinapippi — CHANGES_REQUESTED** [review]
Nice implementation of the vertical profile. Mostly style changes needed.

Critical change: whenever the Timeseries view is selected and the elevation bar is used it switches back to the vertical profile view, this should not be the case.

File: src/components/TimeSeriesComponent/ElevationChart.vue [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+interface Tag {
+  id: string;
+  name: string;
+  disabled: boolean;
+  legendSvg: string;
+}
+
+@Component
+export default class ElevationChart extends Vue {
+  @Prop({
```
[the remark this answers is not in the document]
Comment (vanmeeuwen) [reply src/components/TimeSeriesComponent/ElevationChart.vue:38]: Going to replace when moving to Vue3 where props are handled differently

File: src/components/TimeSeriesComponent/ElevationChart.vue [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+
+  toggleLine(id: string): void {
+    const tag = this.legendTags.find(tag => {
+      return tag.id === id
+    })
+    if (tag) {
+      tag.disabled = !tag.disabled
+    }
+
+    toggleChartVisisbility(this.axis, id)
```
[the remark this answers is not in the document]
Comment (vanmeeuwen) [reply src/components/TimeSeriesComponent/ElevationChart.vue:280]: Typo is in the web-oc-charts. I will open a ticket there.

File: src/components/TimeSeriesComponent/ElevationChart.vue [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+  beforeDestroy(): void {
+    window.removeEventListener('resize', this.resize)
+  }
+
+  get fullscreenIcon(): string {
+    return this.isFullscreen ? 'mdi-fullscreen-exit' : 'mdi-fullscreen'
+  }
+
+  setChartConfigValues(axisOptions: CartesianAxesOptions): void {
+    const chartConfig = this.value
```
[the remark this answers is not in the document]
Comment (vanmeeuwen) [reply src/components/TimeSeriesComponent/ElevationChart.vue:120]: This assignment is there to not use this.value within this function, but the more descriptive chartConfig. Will be fixed when using Vue 3

File: src/components/TimeSeriesComponent/ElevationChart.vue (hunk 1 above)
Comment (vanmeeuwen) [reply src/components/TimeSeriesComponent/ElevationChart.vue:147]: done

**Pull request merged**